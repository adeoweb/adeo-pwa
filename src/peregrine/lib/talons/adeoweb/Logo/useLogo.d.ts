type TUseLogoProps = {
    logoPathPrefix?: string;
};

export type TUseLogo = {
    storeName: string;
    logoSrc: string;
    logoAlt: string;
    logoHeight?: number;
    logoWidth?: number;
};

export function useLogo(props: TUseLogoProps): TUseLogo;
