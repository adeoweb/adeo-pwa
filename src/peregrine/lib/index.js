// components
export { default as PeregrineContextProvider } from './PeregrineContextProvider';
