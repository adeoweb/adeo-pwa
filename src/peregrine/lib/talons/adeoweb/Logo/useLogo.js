import { useConfigContext } from '../../../context/adeoweb/config';

export const useLogo = props => {
    const { logoPathPrefix } = props;
    const [
        {
            store_name: storeName,
            header_logo_src: headerLogoSrc,
            logo_alt: logoAlt,
            logo_height: logoHeight,
            logo_width: logoWidth
        }
    ] = useConfigContext();
    const logoSrc = [logoPathPrefix, headerLogoSrc].join('');

    return {
        storeName: storeName || '',
        logoSrc,
        logoAlt: logoAlt || '',
        logoHeight: logoHeight || undefined,
        logoWidth: logoWidth || undefined
    };
};
