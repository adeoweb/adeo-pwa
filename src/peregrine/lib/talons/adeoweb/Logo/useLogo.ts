import { useConfigContext } from '../../../context/adeoweb/config';

type TUseLogoProps = {
    logoPathPrefix?: string;
};

export type TUseLogo = {
    storeName: string;
    logoSrc: string;
    logoAlt: string;
    logoHeight?: number;
    logoWidth?: number;
};

export const useLogo = (props: TUseLogoProps): TUseLogo => {
    const { logoPathPrefix } = props;
    const [
        {
            store_name: storeName,
            header_logo_src: headerLogoSrc,
            logo_alt: logoAlt,
            logo_height: logoHeight,
            logo_width: logoWidth
        }
    ] = useConfigContext();
    const logoSrc = [logoPathPrefix, headerLogoSrc].join('');

    return {
        storeName: storeName ?? '',
        logoSrc,
        logoAlt: logoAlt ?? '',
        logoHeight: logoHeight ?? undefined,
        logoWidth: logoWidth ?? undefined
    };
};
