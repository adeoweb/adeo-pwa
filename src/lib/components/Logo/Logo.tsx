import React, { FunctionComponent } from 'react';

import Image from 'src/lib/components/Image';
import {
    DEFAULT_LOGO_HEIGHT,
    DEFAULT_LOGO_WIDTH,
    LOGO_PATH_PREFIX
} from 'src/lib/constants/logo';
import { useLogo } from 'src/peregrine/lib/talons/adeoweb/Logo/useLogo';

const Logo: FunctionComponent = () => {
    const { storeName, logoSrc, logoAlt, logoHeight, logoWidth } = useLogo({
        logoPathPrefix: LOGO_PATH_PREFIX
    });

    if (!logoSrc) {
        return null;
    }

    return (
        <Image
            src={logoSrc}
            height={logoHeight || DEFAULT_LOGO_HEIGHT}
            width={logoWidth || DEFAULT_LOGO_WIDTH}
            alt={logoAlt}
            title={storeName}
        />
    );
};

export default Logo;
