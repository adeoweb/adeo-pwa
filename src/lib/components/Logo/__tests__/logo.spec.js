import React from 'react';

import { createTestInstance } from '@magento/peregrine';

import Logo from '../Logo';

jest.mock('src/peregrine/lib/talons/adeoweb/Logo/useLogo', () => {
    return {
        useLogo: () => ({
            storeName: 'Store Name',
            logoSrc: 'https://header.logo/src.jpg',
            logoAlt: 'logo',
            logoHeight: 50,
            logoWidth: 50
        })
    };
});

test('renders correctly', () => {
    const tree = createTestInstance(<Logo />);

    expect(tree.toJSON()).toMatchSnapshot();
});
