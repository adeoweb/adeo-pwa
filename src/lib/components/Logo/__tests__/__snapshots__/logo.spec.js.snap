// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="undefined undefined"
>
  <img
    alt="logo"
    className="undefined undefined"
    height={50}
    loading="eager"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII="
    style={
      Object {
        "height": "50px",
        "width": "50px",
      }
    }
    title="Store Name"
    width={50}
  />
  <img
    alt="logo"
    className="undefined undefined"
    height={50}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="https://header.logo/src.jpg"
    style={
      Object {
        "height": "50px",
        "width": "50px",
      }
    }
    title="Store Name"
    width={50}
  />
</div>
`;
